matrix/common/etl/transformers/cell_expression.py

```diff
@@ -165,9 +174,108 @@ def _parse_10x_bundle(self, bundle_dir):
 
         return cell_lines, expression_lines
 
+    def _parse_optimus_bundle(self, bundle_dir, test_n_cells: int=0):
+        """
+        Parses optimus analysis files into PSV rows for cell and expression Redshift tables.
+        """
+        keys = self._parse_keys(bundle_dir)
+
+        # read expression matrix from zarr
+        store = DCPZarrStore(bundle_dir=bundle_dir)
+        root = zarr.group(store=store)
+
+        n_cells = root.expression_matrix.cell_id.shape[0] if not test_n_cells else test_n_cells
+        chunk_size = root.expression_matrix.expression.chunks[0] if not test_n_cells else test_n_cells - 1
+        n_chunks = int(math.ceil(n_cells / chunk_size))
+
+        cell_lines = set()
+        expression_lines = []
+        for i in range(n_chunks):
+            self._parse_optimus_chunk(
+                keys=keys,
+                root=root,
+                start_row=chunk_size * i,
+                end_row=(i + 1) * chunk_size if (i + 1) * chunk_size < n_cells else n_cells,
+                cell_lines=cell_lines,
+                expression_lines=expression_lines
+            )
+
+        return cell_lines, expression_lines
+
+    def _parse_optimus_chunk(self,
+                             keys: dict,
+                             root: zarr.Group,
+                             start_row: int,
+                             end_row: int,
+                             cell_lines: set,
+                             expression_lines: list):
+        """
+        Parses a chunk of an expression matrix into cell and expression PSV lines.
+        Modifies cell_lines and expression_lines.
+        :param keys: Metadata keys generated by _parse_keys
+        :param root: Zarr group of the full expression matrix
+        :param start_row: Start row of the chunk
+        :param end_row: End row of the chunk
+        :param cell_lines: Output cell PSV lines
+        :param expression_lines: Output expression PSV lines
+        """
+        chunk_size = end_row - start_row
+        n_genes = root.expression_matrix.gene_id.shape[0]
+        expr_values = root.expression_matrix.expression[start_row:end_row]
+        barcodes = root.expression_matrix.cell_id[start_row:end_row]
+
+        for i in range(chunk_size):
+            cell_key = self._generate_10x_cell_key(keys["bundle_uuid"], barcodes[i])
+            gene_count = numpy.count_nonzero(expr_values[i])
+            cell_line = '|'.join(
+                [cell_key,
+                 keys["cell_suspension_key"],
+                 keys["project_key"],
+                 keys["specimen_key"],
+                 keys["library_key"],
+                 keys["analysis_key"],
+                 barcodes[i],
+                 str(gene_count)]
+            ) + '\n'
+            cell_lines.add(cell_line)
+
+            # skip 0 counts
+            if gene_count == 0:
```

Comment:
Does this happen? It would be a weird thing to see in the optimus result I think.

Reply:
Oh doesn't look like it does. Removing